plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPlugin.kt

```diff
@@ -107,11 +108,7 @@ private fun Project.javaVersion(): JavaVersion {
   return JavaVersion.current()
 }
 
-private fun Project.findDokkaTask(): String {
+private fun Project.findDokkaTask(): Provider<String> = provider {
   val tasks = project.tasks.withType(DokkaTask::class.java)
-  return if (tasks.size == 1) {
-    tasks.first().name
-  } else {
-    tasks.findByName("dokkaHtml")?.name ?: "dokka"
-  }
+  tasks.singleOrNull()?.name ?: "dokkaHtml"
```

Comment:
The `"dokka"` fallback was from the 0.x dokka versions. It's been ages since the default task is `dokkaHtml` and those older versions of dokka are probably not even compatible with out min supported Kotlin version.